Add 'from' option for release command

bnigh opened the pull request:
# What Changed
- Added `from` option to `auto release` command

# Why
- See #534

Todo:
- [ ] Add tests
- [x] Add docs
- [ ] Add yourself to contributors (run `yarn contributors:add`)

bnigh commented:
Tested locally, but didn't added any unit tests.
I didn't see any existing test for `auto release` with options (like `--use-version`). Is there a specific place I should add tests, or is above good for pr if existing tests are passing?

hipstersmoothie commented:
This looks good! The tests for this function are [here](https://github.com/intuit/auto/blob/master/packages/core/src/__tests__/auto.test.ts#L732) You should be able to add one to cover this new branch.

Just add a test and add yourself to the contributors table and I'll merge

bnigh commented:
> This looks good! The tests for this function are [here](https://github.com/intuit/auto/blob/master/packages/core/src/__tests__/auto.test.ts#L732) You should be able to add one to cover this new branch.

Addressed in 20ee298

> Just add a test and add yourself to the contributors table and I'll merge

Addressed in d77f50f
